### PR TITLE
Add merge_algorithm to system.merges

### PR DESCRIPTION
**Pull request by amosbird**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Add `merge_algorithm` to system.merges table to improve merging inspections.

Detailed description / Documentation draft:

system.merges documentation is updated.

### CODE REVIEW

File: src/Storages/MergeTree/MergeAlgorithm.cpp [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+{
+    extern const int NOT_IMPLEMENTED;
+}
+
+String toString(MergeAlgorithm merge_algorithm)
+{
+    switch (merge_algorithm)
+    {
+        case MergeAlgorithm::Horizontal:
+            return "HORIZONTAL";
```
Comment (alexey-milovidov) [inline src/Storages/MergeTree/MergeAlgorithm.cpp:16]: Let's make it in the same style as in enum:
Horizontal, Vertical.

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
LGTM and one change requested.

**Review by alexey-milovidov — APPROVED** [review]
The failure of "Yandex synchronization check" is because `MergeAlgorithm.cpp` is not in ya.make.
We should fix it and ya.make.in also.

**Comment by alesapin** [comment]
```
E           Code: 48. DB::Exception: Received from 172.19.0.5:9000. DB::Exception: Unknown MergeAlgorithm 18446744072614756030. Stack trace:
E           
E           0. /build/obj-x86_64-linux-gnu/../contrib/libcxx/include/exception:129: Poco::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int) @ 0x2fc0ac29 in /usr/bin/clickhouse
E           1. /build/obj-x86_64-linux-gnu/../src/Common/Exception.cpp:38: DB::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int) @ 0x108331ee in /usr/bin/clickhouse
E           2. /build/obj-x86_64-linux-gnu/../contrib/libcxx/include/string:1426: DB::Exception::Exception<char const (&) [26], unsigned long>(int, char const (&) [26], unsigned long&&) @ 0x28e4ebbc in /usr/bin/clickhouse
E           3. /build/obj-x86_64-linux-gnu/../src/Storages/MergeTree/MergeAlgorithm.cpp:21: DB::toString(DB::MergeAlgorithm) @ 0x28e4e9c3 in /usr/bin/clickhouse
E           4. /build/obj-x86_64-linux-gnu/../src/Storages/MergeTree/MergeList.cpp:77: DB::MergeListElement::getInfo() const @ 0x28e4df3c in /usr/bin/clickhouse
E           5. /build/obj-x86_64-linux-gnu/../contrib/libcxx/include/new:253: DB::MergeList::get() const @ 0x267ab497 in /usr/bin/clickhouse
E           6. /build/obj-x86_64-linux-gnu/../contrib/libcxx/include/list:618: DB::StorageSystemMerges::fillData(std::__1::vector<COW<DB::IColumn>::mutable_ptr<DB::IColumn>, std::__1::allocator<COW<DB::IColumn>::mutable_ptr<DB::IColumn> > >&, DB::Context const&, DB::SelectQueryInfo const&) const @ 0x267a8ae9 in /usr/bin/clickhouse
E           7. /build/obj-x86_64-linux-gnu/../contrib/libcxx/include/vector:656: DB::IStorageSystemOneBlock<DB::StorageSystemMerges>::read(std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, std::__1::shared_ptr<DB::StorageInMemoryMetadata const> const&, DB::SelectQueryInfo const&, DB::Context const&, DB::QueryProcessingStage::Enum, unsigned long, unsigned int) @ 0x2670473c in /usr/bin/clickhouse
E           8. /build/obj-x86_64-linux-gnu/../src/Processors/QueryPlan/ReadFromStorageStep.cpp:39: DB::ReadFromStorageStep::ReadFromStorageStep(std::__1::shared_ptr<DB::RWLockImpl::LockHolderImpl>, std::__1::shared_ptr<DB::StorageInMemoryMetadata const>&, DB::SelectQueryOptions, std::__1::shared_ptr<DB::IStorage>, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, DB::SelectQueryInfo const&, std::__1::shared_ptr<DB::Context>, DB::QueryProcessingStage::Enum, unsigned long, unsigned long) @ 0x29a54168 in /usr/bin/clickhouse
E           9. /build/obj-x86_64-linux-gnu/../contrib/libcxx/include/memory:3028: std::__1::__unique_if<DB::ReadFromStorageStep>::__unique_single std::__1::make_unique<DB::ReadFromStorageStep, std::__1::shared_ptr<DB::RWLockImpl::LockHolderImpl>&, std::__1::shared_ptr<DB::StorageInMemoryMetadata const>&, DB::SelectQueryOptions&, std::__1::shared_ptr<DB::IStorage>&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > >&, DB::SelectQueryInfo&, std::__1::shared_ptr<DB::Context>&, DB::QueryProcessingStage::Enum&, unsigned long&, unsigned long&>(std::__1::shared_ptr<DB::RWLockImpl::LockHolderImpl>&, std::__1::shared_ptr<DB::StorageInMemoryMetadata const>&, DB::SelectQueryOptions&, std::__1::shared_ptr<DB::IStorage>&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > >&, DB::SelectQueryInfo&, std::__1::shared_ptr<DB::Context>&, DB::QueryProcessingStage::Enum&, unsigned long&, unsigned long&) @ 0x27c9f321 in /usr/bin/clickhouse
E           10. /build/obj-x86_64-linux-gnu/../src/Interpreters/InterpreterSelectQuery.cpp:1412: DB::InterpreterSelectQuery::executeFetchColumns(DB::QueryProcessingStage::Enum, DB::QueryPlan&, std::__1::shared_ptr<DB::PrewhereInfo> const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) @ 0x27c779ab in /usr/bin/clickhouse
E           11. /build/obj-x86_64-linux-gnu/../src/Interpreters/InterpreterSelectQuery.cpp:0: DB::InterpreterSelectQuery::executeImpl(DB::QueryPlan&, std::__1::shared_ptr<DB::IBlockInputStream> const&, std::__1::optional<DB::Pipe>) @ 0x27c68ab7 in /usr/bin/clickhouse
E           12. /build/obj-x86_64-linux-gnu/../contrib/libcxx/include/optional:224: DB::InterpreterSelectQuery::buildQueryPlan(DB::QueryPlan&) @ 0x27c672da in /usr/bin/clickhouse
E           13. /build/obj-x86_64-linux-gnu/../src/Interpreters/InterpreterSelectWithUnionQuery.cpp:0: DB::InterpreterSelectWithUnionQuery::buildQueryPlan(DB::QueryPlan&) @ 0x28102b31 in /usr/bin/clickhouse
E           14. /build/obj-x86_64-linux-gnu/../src/Interpreters/InterpreterSelectWithUnionQuery.cpp:0: DB::InterpreterSelectWithUnionQuery::execute() @ 0x28103f8a in /usr/bin/clickhouse
E           15. /build/obj-x86_64-linux-gnu/../src/Interpreters/executeQuery.cpp:392: DB::executeQueryImpl(char const*, char const*, DB::Context&, bool, DB::QueryProcessingStage::Enum, bool, DB::ReadBuffer*) @ 0x2842e00c in /usr/bin/clickhouse
E           16. /build/obj-x86_64-linux-gnu/../src/Interpreters/executeQuery.cpp:679: DB::executeQuery(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, DB::Context&, bool, DB::QueryProcessingStage::Enum, bool) @ 0x2842b737 in /usr/bin/clickhouse
E           17. /build/obj-x86_64-linux-gnu/../src/Server/TCPHandler.cpp:0: DB::TCPHandler::runImpl() @ 0x2943ed9c in /usr/bin/clickhouse
E           18. /build/obj-x86_64-linux-gnu/../src/Server/TCPHandler.cpp:0: DB::TCPHandler::run() @ 0x2945be6b in /usr/bin/clickhouse
E           19. /build/obj-x86_64-linux-gnu/../contrib/poco/Net/src/TCPServerConnection.cpp:57: Poco::Net::TCPServerConnection::start() @ 0x2fa608cf in /usr/bin/clickhouse
E           20. /build/obj-x86_64-linux-gnu/../contrib/libcxx/include/atomic:1036: Poco::Net::TCPServerDispatcher::run() @ 0x2fa615d2 in /usr/bin/clickhouse
E           21. /build/obj-x86_64-linux-gnu/../contrib/poco/Foundation/src/ThreadPool.cpp:0: Poco::PooledThread::run() @ 0x2fd2f926 in /usr/bin/clickhouse
E           22. /build/obj-x86_64-linux-gnu/../contrib/poco/Foundation/src/Thread_POSIX.cpp:0: Poco::ThreadImpl::runnableEntry(void*) @ 0x2fd29afe in /usr/bin/clickhouse
E           23. start_thread @ 0x9669 in /usr/lib/x86_64-linux-gnu/libpthread-2.30.so
E           24. clone @ 0x1222b3 in /usr/lib/x86_64-linux-gnu/libc-2.30.so
```
Failure looks related to changes.

**Comment by amosbird** [comment]
> Failure looks related to changes.

Yes. And it's fixed.

**Comment by alesapin** [comment]
AST fuzzer reported https://github.com/ClickHouse/ClickHouse/issues/14700, performance not related.

**Review by alesapin — APPROVED** [review]

**Pull request merged**